### PR TITLE
Use named-js-regexp instead of xregexp

### PR DESCRIPTION
**Pull request by steelbrain**
cc @Arcanemagus 

Pros
 - It's backward compatible
 - It's lightweight, so the tons of packages depending on this package that don't use it will enjoy it

Cons
 - Not yet battle tested on our scale (it does have a test suite)

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/atom-linter/98)

### CODE REVIEW

**Comment by steelbrain** [comment]
@edvinv before I merge this, I need your confirmation that you will follow semver with this package and not break it's compatibility in the next update

File: lib/helpers.js [hunk 1]
```diff
@@ -246,14 +246,36 @@ function find(directory, name) {
     if (currentDir === '') {
       currentDir = Path.resolve(directory, '/');
     }
-    for (const fileName of names) {
-      const filePath = Path.join(currentDir, fileName);
+    var _iteratorNormalCompletion = true;
```
Comment (steelbrain) [inline lib/helpers.js]: All of this unrelated stuff is from babel, it's optimizing the `for...of` loop

**Comment by edvinv** [comment]
@steelbrain 
Semver is strictly followed, but anyway there are no plans to add any incompatibility feature.

**Comment by steelbrain** [comment]
Published as v4.3.3
